components/calendar/calendar.js

```diff
@@ -0,0 +1,488 @@
+					}
+					numDaysChange *= -1;
+				} else {
+					numDaysChange = 6 - dayOfTheWeek + firstDayOfWeek;
+					if (numDaysChange > 6) {
+						numDaysChange -= daysInWeek;
+					}
+				}
+				preventDefault = true;
+				break;
+			} case keyCodes.PAGEUP: {
+				const diff = getNumberOfDaysToSameWeekPrevMonth(this._shownMonth, this._shownYear);
+
+				this._focusDate.setDate(this._focusDate.getDate() - diff);
+				this._keyboardTriggeredMonthChange = true;
+
+				// handle when current month has more weeks than previous month and page up pressed from last week
+				const newFocusDateMonth = this._focusDate.getMonth();
+				if (newFocusDateMonth === this._shownMonth) this._focusDate.setDate(this._focusDate.getDate() - 7);
+
+				this._focusDate = new Date(this._focusDate.getFullYear(), this._focusDate.getMonth(), this._focusDate.getDate());
+
+				this._updateShownMonthDecrease();
+				preventDefault = true;
+				break;
+			} case keyCodes.PAGEDOWN: {
+				const nextMonth = getNextMonth(this._shownMonth);
+				const diff = getNumberOfDaysToSameWeekPrevMonth(nextMonth, this._shownYear);
+
+				this._focusDate.setDate(this._focusDate.getDate() + diff);
+				this._keyboardTriggeredMonthChange = true;
+
+				// handle when current month has more weeks than next month and page down pressed from last week
+				const newFocusDateMonth = this._focusDate.getMonth();
+				if ((newFocusDateMonth - this._shownMonth) > 1
+					|| (newFocusDateMonth === 1 && this._shownMonth === 11)
+				) {
+					this._focusDate.setDate(this._focusDate.getDate() - 7);
+				}
+				this._focusDate = new Date(this._focusDate.getFullYear(), this._focusDate.getMonth(), this._focusDate.getDate());
+				this._updateShownMonthIncrease();
+				preventDefault = true;
+				break;
+			}
+		}
+
+		if (numDaysChange) {
+			this._updateFocusDateOnKeyDown(numDaysChange);
+		}
+
+		if (preventDefault) {
+			e.preventDefault();
+			e.stopPropagation();
+		}
+	}
+
+	_onNextMonthButtonClick() {
+		this._updateShownMonthIncrease();
+		this._updateFocusDateOnMonthButtonClick();
+	}
+
+	_onPrevMonthButtonClick() {
+		this._updateShownMonthDecrease();
+		this._updateFocusDateOnMonthButtonClick();
+	}
+
+	_updateFocusDateOnMonthButtonClick() {
+		// if selectedValue is in the month, that should be the focus date,
+		// else the 1st of the month should be
+		const selectedValueDate = this.selectedValue ? parseDate(this.selectedValue) : null;
+		if (selectedValueDate && selectedValueDate.getMonth() === this._shownMonth) {
+			this._focusDate = new Date(selectedValueDate.getFullYear(), selectedValueDate.getMonth(), selectedValueDate.getDate());
+		} else {
+			this._focusDate = new Date(this._shownYear, this._shownMonth, 1);
+		}
+	}
+
+	_updateFocusDateOnKeyDown(numDays) {
+		const oldFocusDate = new Date(this._focusDate);
+		this._focusDate.setDate(this._focusDate.getDate() + numDays);
+		this._focusDate = new Date(this._focusDate.getFullYear(), this._focusDate.getMonth(), this._focusDate.getDate());
```

Comment:
Nit: I might combine these 2 lines so you're not changing the date value and then replacing the object one after the other:

    this._focusDate = new Date(
        this._focusDate.getFullYear(),
        this._focusDate.getMonth(),
        this._focusDate.getDate() + numDays
    );